src/main/java/gregtech/common/items/behaviors/ItemMagnetBehavior.java

```diff
@@ -0,0 +1,132 @@
+package gregtech.common.items.behaviors;
+
+import gregtech.api.capability.GregtechCapabilities;
+import gregtech.api.capability.IElectricItem;
+import gregtech.api.items.metaitem.MetaItem;
+import gregtech.api.items.metaitem.stats.IItemBehaviour;
+import net.minecraft.client.resources.I18n;
+import net.minecraft.entity.Entity;
+import net.minecraft.entity.item.EntityItem;
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.ActionResult;
+import net.minecraft.util.EnumActionResult;
+import net.minecraft.util.EnumHand;
+import net.minecraft.util.math.AxisAlignedBB;
+import net.minecraft.util.math.MathHelper;
+import net.minecraft.util.text.TextComponentTranslation;
+import net.minecraft.world.World;
+import net.minecraftforge.common.MinecraftForge;
+import net.minecraftforge.event.entity.item.ItemTossEvent;
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent;
+import net.minecraftforge.items.ItemHandlerHelper;
+import net.minecraftforge.items.ItemStackHandler;
+
+import javax.annotation.Nonnull;
+import java.util.List;
+
+public class ItemMagnetBehavior implements IItemBehaviour {
+
+    private final int range;
+    private final float speed;
+    private boolean isActive;
+
+    public ItemMagnetBehavior(int range, float speed) {
+        this.range = range;
+        this.speed = speed;
+        MinecraftForge.EVENT_BUS.register(this);
+    }
+
+    @Override
+    public ActionResult<ItemStack> onItemRightClick(World world, @Nonnull EntityPlayer player, EnumHand hand) {
+        if (!player.world.isRemote) {
+            this.isActive = !this.isActive;
+            player.sendMessage(new TextComponentTranslation(isActive ? "behavior.item_magnet.enabled" : "behavior.item_magnet.disabled"));
+        }
+        return ActionResult.newResult(EnumActionResult.PASS, player.getHeldItem(hand));
+    }
+
+    @Override
+    public void onUpdate(ItemStack itemStack, Entity entity) {
+        if (!isActive || !(entity instanceof EntityPlayer) || entity.getEntityWorld().isRemote)
+            return;
+
+        EntityPlayer entityPlayer = (EntityPlayer) entity;
+        List<EntityItem> itemsInRange = entityPlayer.getEntityWorld().getEntitiesWithinAABB(EntityItem.class, getAreaBoundingBox(entityPlayer));
+        for (EntityItem entityItem : itemsInRange) {
+            if (entityItem.isDead) continue;
+
+            double distanceX = (entityPlayer.getPosition().getX() + 0.5) - entityItem.posX;
+            double distanceY = (entityPlayer.getPosition().getY() + 0.5) - entityItem.posY;
+            double distanceZ = (entityPlayer.getPosition().getZ() + 0.5) - entityItem.posZ;
+            double distance = MathHelper.sqrt(distanceX * distanceX + distanceZ * distanceZ);
+            if (distance >= 0.7) {
+                if (!entityItem.cannotPickup()) {
+                    if (!drainEnergy(true, itemStack, (int) distance))
```

Comment:
Not that it really matters, but why are you casting to `int` when the call takes a `long`?